File: rag-service/frontend/src/documents/uploadResult.ts
import type { UploadResult } from '@/types/rag'

export function uploadAction(result: UploadResult): 'OPEN_EXISTING' | 'WATCH_JOB' {
  return result.duplicate || result.jobId == null ? 'OPEN_EXISTING' : 'WATCH_JOB'
}

File: rag-service/backend/src/main/java/com/agentto/rag/document/UploadResult.java
package com.agentto.rag.document;

public record UploadResult(Long documentId, Long versionId, Long jobId, String objectKey,
        boolean duplicate, String message) {

    public static UploadResult created(Long documentId, Long versionId, Long jobId, String objectKey) {
        return new UploadResult(documentId, versionId, jobId, objectKey, false, "文件已进入处理队列");
    }

    public static UploadResult duplicate(RagDocumentVersion version) {
        return new UploadResult(version.getDocumentId(), version.getId(), null, version.getObjectKey(), true,
                "该文件已经入库，已为你打开已有文档");
    }
}
